### PR TITLE
User docs

### PR DESCRIPTION
**Pull request by sithlord48**
split from #7673 
Pretty bare bones but its a start 

 - New [`doc/configuration.md`](https://github.com/deskflow/deskflow/blob/user_docs/doc/configuration.md)
    - Readme for  config files and examples
    - move all existing examples here remove old files
- rm `BUILD.MD`
    - Focus is on user documents the build info is in the wiki and we should point to that
- rm `CONTRIBUTING.md`
    - Focus is on user documents the build info is in the wiki and we should point to that
- New [`doc/mainpage.md`](https://github.com/deskflow/deskflow/blob/user_docs/doc/mainpage.md)
- New `CMakeLists.txt` to generate documents.

### CODE REVIEW

File: doc/mainpage.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+## Configuration
+Our [Configuration] page has example configurations
+
+
+## More Info on our Wiki
+
+https://github.com/deskflow/deskflow/wiki
+
+[Configuration]:configuration.md
```
Comment (nbolton) [inline doc/mainpage.md:16]: What does this do?

File: doc/mainpage.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+**Deskflow** is a free and open source keyboard and mouse sharing app.
+Use the keyboard, mouse, or trackpad of one computer to control nearby computers,
+and work seamlessly between them.
+
+
+## Configuration
+Our [Configuration] page has example configurations
+
+
+## More Info on our Wiki
```
Comment (nbolton) [inline doc/mainpage.md:12]:
```suggestion
## More info
```

File: doc/mainpage.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+https://github.com/deskflow/deskflow/wiki
```
Comment (nbolton) [inline doc/mainpage.md:14]:
```suggestion
For more info, see our [Wiki](https://github.com/deskflow/deskflow/wiki).
```

File: doc/configuration.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+[client.args]
+no-daemon = true
+no-tray = true
+debug = "DEBUG2"
+name = "larry"
+_last = "moe:24800"
+```
+
+
+# Example Environment
```
Comment (nbolton) [inline doc/configuration.md:146]:
```suggestion
# Example `.env` file
```

File: doc/configuration.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+		left  = moe
+		up    = curly
+
+	# larry is below curly.
+	curly:
+		down  = larry
+end
+```
+
+# Example file for `--config-toml` arg.
```
Comment (nbolton) [inline doc/configuration.md:127]:
```suggestion
# Example file for `--config-toml` arg
```

File: doc/configuration.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+no-tray = true
+debug = "DEBUG2"
+name = "larry"
+_last = "moe:24800"
+```
+
+
+# Example Environment
+
+Example contents for the .env file
```
Comment (nbolton) [inline doc/configuration.md:148]:
```suggestion
```

**Review by nbolton — DISMISSED** [review]
Approved but minor nits you may want to fix

File: doc/mainpage.md (hunk 1 above)
Comment (sithlord48) [reply doc/mainpage.md:16]: It makes [foo]  everywhere in the md a link to :bar

**Review by nbolton — APPROVED** [review]

**Pull request merged**